src/main/java/springboot/webapp/usersmanager/repositories/UserRepository.java

```diff
@@ -0,0 +1,18 @@
+package springboot.webapp.usersmanager.repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import springboot.webapp.usersmanager.entities.ERole;
+import springboot.webapp.usersmanager.entities.User;
+
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Integer> {
+    User findById(int id);
+
+    @Modifying
+    @Query("update User u set u.name = ?1, u.surname = ?2, u.email =?3, u.role = ?4 where u.id = ?5")
+    void setUserInfoById(String name, String surname, String email, ERole role, Integer userId);
```

Comment:
there is default method for updating in JPA

Reply:
could not find it

Reply:
use Google
https://stackoverflow.com/questions/11881479/how-do-i-update-an-entity-using-spring-data-jpa

Reply:
Should i use this example?

public void updateUser(Userinfos u) {
    User userFromDb = userRepository.findById(u.getid());
    // crush the variables of the object found
    userFromDb.setFirstname("john"); 
    userFromDb.setLastname("dew");
    userFromDb.setAge(16);
    userRepository.save(userFromDb);
}

//My solution from this article too. And i see two ways: implement own update method or use JPA save method, but i think this is adds to much code